application/controllers/Hook.php

```diff
@@ -0,0 +1,254 @@
+<?php
+defined('BASEPATH') OR exit('No direct script access allowed');
+
+class Hook extends MY_Base {
+
+    private $execution_id = null;
+    private $pattern = '/[0-9]{4}-[0-9]{2}-[0-9]{2}-(.*)?\.json/';
+    private $force = false;
+
+    public function add()
+    {
+        $this->load->model('Execution');
+        $this->load->model('Suite');
+        $this->load->model('Test');
+
+        log_message('info', "verifying data");
+        if (!$this->input->get('token') || !$this->input->get('filename')) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("no enough parameters");
```

Comment:
```suggestion
            exit(json_encode(['error' => "no enough parameters"])); 
```

Reply:
That's a real question here: do I need to `json_encode` all my error messages or just the ones after retrieving the data ?

Reply:
Soooo I encoded them all.